### PR TITLE
[online] Revert GENERATOR_HOST explicit setting

### PR DESCRIPTION
**Pull request by jimschubert**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`. If contributing template-only or documentation-only changes which will change sample output, be sure to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) first.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.1.x`, `5.0.x`. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

Removing `GENERATOR_HOST` usage, and moving configuration to server.  We may want to consider removing `GENERATOR_HOST` completely from GenApiServer (if possible), and just using built-in support for retrieving the host/port. I was trying to reduce confusion around this variable in the code that is removed by this PR, but upon discussion it seems the environment variable just adds confusion and is therefore not the appropriate configuration to target.

### CODE REVIEW

**Comment by jimschubert** [comment]
@cbornet @wing328

File: modules/openapi-generator-online/Dockerfile [hunk 1]
```diff
@@ -4,9 +4,10 @@ WORKDIR /generator
 
 COPY target/openapi-generator-online.jar /generator/openapi-generator-online.jar
 
-ENV GENERATOR_HOST=http://localhost
+# GENERATOR_HOST can be used to determine the target location of a download link.
+# The default value asumes binding to host via: docker -p 8080:8080 image_name
+ENV GENERATOR_HOST=http://localhost:8080
```
Comment (cbornet) [inline modules/openapi-generator-online/Dockerfile]: I'm not sure I understand. Do we keep GENERATOR_HOST?

Comment (jimschubert) [reply modules/openapi-generator-online/Dockerfile]: @cbornet `GENERATOR_HOST` is still used here:

https://github.com/OpenAPITools/openapi-generator/blob/109808e60d15e97680f274bda8ed854da52b67ea/modules/openapi-generator-online/src/main/java/org/openapitools/codegen/online/service/GenApiService.java#L161

This isn't "correct" and will need to be changed. The previous value here would be incorrect for all invocations unless bound to port 80, which I believe requires super user privileges. This change at least causes the variable to match the exposed port.

Many people try out docker images by binding to random host ports (`-P`), and when doing this… `GENERATOR_HOST` will be useless and the file download link generation won't work.

This PR limits focus to get the proxying host setting working on the server. I can look at the other work later.

**Comment by jimschubert** [comment]
This reverts changes I added in #3263 and #3265.

**Comment by jimschubert** [comment]
CircleCI failure is unrelated. Merging this to let the deployment do its thing, and creating an issue for tracking removal of `GENERATOR_HOST` from GenApiService.

File: modules/openapi-generator-online/Dockerfile (hunk 1 above)
Comment (jimschubert) [reply modules/openapi-generator-online/Dockerfile]: Created #3288 to address this other concern.

**Pull request merged**